### DISCUSSION TITLE
sysdate as an AnsiFunction

### DISCUSSION THREAD

**Discussion by csdev** [sqlalchemy/sqlalchemy#8003] [Usage Questions] [answered]
I noticed that SQLAlchemy implements `sysdate` as an `AnsiFunction`, which means that it does not render parentheses. However, some databases like Snowflake [require the parentheses](https://docs.snowflake.com/en/sql-reference/functions/sysdate.html#usage-notes). Is it the responsibility of the dialect to override `sysdate` so it works properly, is it the responsibility of the user, or is there a bug in SQLAlchemy for assuming ANSI behavior? (I was able to work around this problem using `text('SYSDATE()')`, but it's not the cleanest solution.)

**Comment by CaselIT** [comment, marked as answer]
> Is it the responsibility of the dialect to override sysdate so it works properly

I would say yes, it should be something that the snowflake should handle. You should probably file an issue on the dialect
